Shared/Extensions/URLExtensions.swift

```diff
@@ -420,6 +420,15 @@ extension URL {
         }
         return ["youtube", "vimeo", "twitch", "twitter", "reddit"].contains(where: domain.contains)
     }
+    
+    // Check if the website is a video streaming content site used inside product notifications
+    public var isVideoSteamingSiteURL: Bool {
+        // Don't need to include Github as it does a page load instead of XHR load.
+        guard let domain = self.baseDomain else {
+            return true
```

Comment:
shouldn't we return false in this case?
You can add a unit test for this helper too

Reply:
Tests are added and domain check is fixed